gpytorch/models/exact_prediction_strategies.py

```diff
@@ -753,39 +774,45 @@ def covar_cache(self):
             constant = torch.tensor(1.0, dtype=lt.dtype, device=lt.device)
 
         train_factor = lt.root.to_dense()
-        train_train_covar = self.lik_train_train_covar
-        inner_term = (
+        train_train_covar = self.lik_train_train_covar.evaluate_kernel()  # HOTFIX here: need to resolve lazy eval!
```

Comment:
@gpleiss what's the deal with this? Why exactly is this necessary? This seems to be causing some other test failure here when running on a GPU, presumably due to changed numerical behavior: https://github.com/meta-pytorch/botorch/blob/main/test/models/test_fully_bayesian.py#L834

Reply:
We need to resolve the LazyEvaluatedKernelTensor, because otherwise the $\boldsymbol K + \sigma^2 I$ matrix will be an `AddedDiagLinearOperator(LazyEvaluatedKernelTensor, DiagLinearOperator)` rather than a `LowRankRootAddedDiagLinearOperator(LowRankRootLinearOperator, DiagLinearOperator)`. The latter is necessary to get the structure we want.

I can remove this `evaluate_kernel` if I add the following line before calling the super constructor:

```python
train_prior_dist = train_prior_dist.__class__(
    train_prior_dist.mean, train_prior_dist.lazy_covariance_matrix.evaluate_kernel()
)
```

(which I originally added after calling the super constructor; moving it before is probably a better idea). I'm not sure if that will fix the numerical issues on the GPU test though?